From e19b50e6d94defbd3569083269bf2b1c89afe830 Mon Sep 17 00:00:00 2001
From: Suvarna Meenakshi <sumeenak@microsoft.com>
Date: Tue, 26 Jul 2022 22:16:24 +0000
Subject: [PATCH 1/2] Allow all incoming traffic on http/https ports for
 restapi Include changes added in PR 3354726 with modification done so that
 multiple threads do not use same config db connector.

Signed-off-by: Suvarna Meenakshi <sumeenak@microsoft.com>
---
 scripts/caclmgrd | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/scripts/caclmgrd b/scripts/caclmgrd
index 19e42a8..38f8619 100755
--- a/scripts/caclmgrd
+++ b/scripts/caclmgrd
@@ -206,6 +206,24 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
         tcp_flags_str = tcp_flags_str[:-1]
         return tcp_flags_str
 
+    def generate_allow_restapi_iptables_commands(self, namespace):
+        # Following ranges from 6 regions are temporarily hardcoded and allowed for Bluebird Service
+        src_ip_range = ["20.44.16.64/27", "13.66.141.96/27", "13.69.229.128/27", "52.162.110.128/27", "52.231.147.224/27", "40.74.146.224/27", "40.78.203.96/27", "20.150.171.224/27", "52.162.110.128/27"]
+
+        dports = ["8081", "8090"]
+
+        allow_restapi_commands = []
+
+        for dport in dports:
+            for src_ip in src_ip_range:
+                ip_ntwrk = ipaddress.ip_network(src_ip, strict=False)
+                if isinstance(ip_ntwrk, ipaddress.IPv4Network):
+                    allow_restapi_commands.append(self.iptables_cmd_ns_prefix[namespace] + "iptables -A INPUT -s {} -p tcp --dport {} -j ACCEPT".format(src_ip, dport))
+                else:
+                    allow_restapi_commands.append(self.iptables_cmd_ns_prefix[namespace] + "ip6tables -A INPUT -s {} -p tcp --dport {} -j ACCEPT".format(src_ip, dport))
+
+        return allow_restapi_commands
+
     def update_feature_present(self):
         feature_tb_info = self.config_db_map[DEFAULT_NAMESPACE].get_table(self.FEATURE_TABLE)
         if feature_tb_info:
@@ -517,6 +535,10 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
         iptables_cmds.append(self.iptables_cmd_ns_prefix[namespace] + "iptables -A INPUT -p tcp --dport 179 -j ACCEPT")
         iptables_cmds.append(self.iptables_cmd_ns_prefix[namespace] + "ip6tables -A INPUT -p tcp --dport 179 -j ACCEPT")
 
+        # Add iptables/ip6tables commands to allow all restapi/http/https requests
+        if "restapi" in self.feature_present and self.feature_present.get("restapi"):
+            iptables_cmds += self.generate_allow_restapi_iptables_commands(namespace)
+
         # Get current ACL tables and rules from Config DB
         self._tables_db_info = self.config_db_map[namespace].get_table(self.ACL_TABLE)
         self._rules_db_info = self.config_db_map[namespace].get_table(self.ACL_RULE)
-- 
2.25.1


From d7866d96544c91add34cd5422cdc780b2447d4f4 Mon Sep 17 00:00:00 2001
From: Sumukha Tumkur Vani <stumkurv@microsoft.com>
Date: Fri, 26 Aug 2022 21:20:58 +0000
Subject: [PATCH 2/2] Add destination IP for allowing outgoing traffic for
 RESTAPI

Signed-off-by: Sumukha Tumkur Vani <stumkurv@microsoft.com>
---
 scripts/caclmgrd | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/scripts/caclmgrd b/scripts/caclmgrd
index 38f8619..a4daae8 100755
--- a/scripts/caclmgrd
+++ b/scripts/caclmgrd
@@ -209,7 +209,7 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
     def generate_allow_restapi_iptables_commands(self, namespace):
         # Following ranges from 6 regions are temporarily hardcoded and allowed for Bluebird Service
         src_ip_range = ["20.44.16.64/27", "13.66.141.96/27", "13.69.229.128/27", "52.162.110.128/27", "52.231.147.224/27", "40.74.146.224/27", "40.78.203.96/27", "20.150.171.224/27", "52.162.110.128/27"]
-
+        dest_ip_range = ["10.20.8.199/32", "10.212.64.1/32", "10.212.64.2/32"]
         dports = ["8081", "8090"]
 
         allow_restapi_commands = []
@@ -222,6 +222,13 @@ class ControlPlaneAclManager(daemon_base.DaemonBase):
                 else:
                     allow_restapi_commands.append(self.iptables_cmd_ns_prefix[namespace] + "ip6tables -A INPUT -s {} -p tcp --dport {} -j ACCEPT".format(src_ip, dport))
 
+            for dest_ip in dest_ip_range:
+                ip_ntwrk = ipaddress.ip_network(dest_ip, strict=False)
+                if isinstance(ip_ntwrk, ipaddress.IPv4Network):
+                    allow_restapi_commands.append(self.iptables_cmd_ns_prefix[namespace] + "iptables -A INPUT -d {} -p tcp --dport {} -j ACCEPT".format(dest_ip, dport))
+                else:
+                    allow_restapi_commands.append(self.iptables_cmd_ns_prefix[namespace] + "ip6tables -A INPUT -d {} -p tcp --dport {} -j ACCEPT".format(dest_ip, dport))
+
         return allow_restapi_commands
 
     def update_feature_present(self):
-- 
2.25.1

